internal/state.go

```diff
@@ -125,6 +119,9 @@ func buildWebSocketURL(engine string, ttl string) string {
 }
 
 func buildMetadataURL(engine string, appID string) string {
+	if appID == "" {
```

Comment:
Should appId ever be an empty string? and can't you get metadata for a sessionApp?